src/debugger/RunDebugAdapter.ts

```diff
@@ -0,0 +1,121 @@
+import * as vscode from "vscode";
+import { DebugProtocol } from "@vscode/debugprotocol";
+
+export class Message implements DebugProtocol.ProtocolMessage {
+  seq: number;
+  type: string;
+
+  public constructor(type: string) {
+    this.seq = 0;
+    this.type = type;
+  }
+}
+
+export class Event extends Message implements DebugProtocol.Event {
+  event: string;
+
+  public constructor(event: string, body?: any) {
+    super("event");
+    this.event = event;
+    if (body) {
+      (<any>this).body = body;
+    }
+  }
+}
+
+class ExitedEvent extends Event implements DebugProtocol.ExitedEvent {
+  body: {
+    exitCode: number;
+  };
+
+  public constructor(exitCode: number) {
+    super("exited");
+    this.body = {
+      exitCode: exitCode,
+    };
+  }
+}
+
+export class TerminatedEvent
+  extends Event
+  implements DebugProtocol.TerminatedEvent
+{
+  public constructor(restart?: any) {
+    super("terminated");
+    if (typeof restart === "boolean" || restart) {
+      const e: DebugProtocol.TerminatedEvent = this;
+      e.body = {
+        restart: restart,
+      };
+    }
+  }
+}
+
+class Response extends Message implements DebugProtocol.Response {
+  request_seq: number;
+  success: boolean;
+  command: string;
+
+  public constructor(request: DebugProtocol.Request, message?: string) {
+    super("response");
+    this.request_seq = request.seq;
+    this.command = request.command;
+    if (message) {
+      this.success = false;
+      (<any>this).message = message;
+    } else {
+      this.success = true;
+    }
+  }
+}
```

Comment:
Everything until here is copied over from https://github.com/microsoft/vscode-debugadapter-node/blob/f33a09029b2d1212f07ba087042a315e8a958c68/adapter/src/messages.ts but I have no idea if we can import it :thinking: